### ISSUE TITLE
["bug"] Add condition statement in Dockerfile

### ISSUE THREAD

**Issue by tongtong1993** [ictr/chime#1]
I updated the codes to v1.1.3 on github and found that the new code no longer use setting.py to store the default parameters, but rather put them all in ./defaults/webapp.cfg and then activated by ./Dockerfile. I have not figured out a way to add condition statements in Dockerfile to activate different hospital’s parameter, do you have any idea?

**Comment by BoPeng** [comment]
The docker file copies the settings into the container through

https://github.com/ictr/chime/blob/fca8e45c2dc9fbaced0ad0667e555bf91940047f/Dockerfile#L8

Since we have a docker compose file with environment variables

https://github.com/ictr/chime/blob/fca8e45c2dc9fbaced0ad0667e555bf91940047f/docker-compose.yml#L10

You can change the docker file to copy site-specific default variables. Something like

```
COPY defaults-$SITE defaults
```

Then, from our source tree, you can have multiple `defaults` directories such as `defaults-Methodist` to store site specific defaults.

**Comment by BoPeng** [comment]
Hold on, a better option might be changing this line of code

https://github.com/ictr/chime/blob/fca8e45c2dc9fbaced0ad0667e555bf91940047f/Dockerfile#L2 

to have something like

```
ENV PARAMETERS=./defaults/webapp-$SITE.cfg 
```
then you only need to have multiple `.cfg` file, instead of multiple `defaults` folder.

**Comment by tongtong1993** [comment]
![image](https://user-images.githubusercontent.com/27018331/79278167-cd0f7280-7e70-11ea-86b7-fb0c9c11544d.png)
It seems docker first read Dockerfile then docker-compose.yml, that's why $SITE is empty.

**Comment by BoPeng** [comment]
Yeah, it seems that `env` is not passed during build time, `args` is. Could you try something like the following, 

```
build:
  args:
    SITE: Methodist
```

Also the variables need to be quoted as `${SITE}` in Dockerfile.

**Comment by tongtong1993** [comment]
`version: '3.1'

services:
  app-tch:
    build:
      context: .
      args:
        SITE: TCH
    restart: always
    ports:
      - "8001:8000"
    environment:
      - SITE=TCH
  app-methodist:
    build:
      context: .
      args:
        SITE: METHODIST
    restart: always
    ports:
      - "8002:8000"
    environment:
      - SITE=METHODIST
  app-bslmc:
    build:
      context: .
      args:
        SITE: BSLMC
    restart: always
    ports:
      - "8003:8000"
    environment:
      - SITE=BSLMC
  app-harrishealth:
    build:
      context: .
      args:
        SITE: HarrisHealth
    restart: always
    ports:
      - "8004:8000"
    environment:
      - SITE=HarrisHealth
  app-houston_va:
    build:
      context: .
      args:
        SITE: HoustonVA
    restart: always
    ports:
      - "8005:8000"
    environment:
      - SITE=HoustonVA
  app-national_va:
    build:
      context: .
      args:
        SITE: NationalVA
    restart: always
    ports:
      - "8006:8000"
    environment:
      - SITE=NationalVA`

With this modified docker-compose.yml, I still get the same error.

**Comment by tongtong1993** [comment]
```
version: '3.1'

services:
  app-tch:
    build:
      context: .
      args:
        SITE: TCH
    restart: always
    ports:
      - "8001:8000"
    environment:
      - SITE=TCH
```
With the modified docker-compose.yml file, I still get the same error.

**Comment by BoPeng** [comment]
You need to read the manual... 

In this case, https://docs.docker.com/compose/compose-file/  says

Add build arguments, which are environment variables accessible only during the build process.

First, specify the arguments in your Dockerfile:
```
ARG buildno
ARG gitcommithash
RUN echo "Build number: $buildno"
RUN echo "Based on commit: $gitcommithash"
```

Then specify the arguments under the build key. You can pass a mapping or a list:

```
build:
  context: .
  args:
    buildno: 1
    gitcommithash: cdc3b19
build:
  context: .
  args:
    - buildno=1
    - gitcommithash=cdc3b19
```
